### DISCUSSION TITLE
Cols property not getting set and text being overwritten after a certain width is occupied

### DISCUSSION THREAD

**Discussion by shiv123-max** [sourcelair/xterm.js#4505] [Q&A] [answered]
Screenshot of HTML Code:

![image](https://user-images.githubusercontent.com/56493775/236137683-40e9eff5-b0d2-4e29-89c9-d75d6940f808.png)

Video of the issue: 

https://user-images.githubusercontent.com/56493775/236138224-230b96e7-6caa-47e2-aa1d-1985445bf9f2.mp4

CSS: 

`.terminal {
    width: 100%;
}`

XTERM VERSION : 5.1.0

**Comment by jerch** [comment]
You also need to tell the PTY about your changed terminal size.

**Comment by shiv123-max** [comment]
I did not get you.
Can you elaborate on that? I am not aware of concepts like PTY or TTY and I didn't understand what changes should I make?

**Comment by jerch** [comment, marked as answer]
From your screencast it looks like you forward real terminal data. This has to come from somewhere, and to have proper terminal semantics with xterm.js you have to go through a PTY on the host system (or conpty with windows). And the size of that PTY has to be in sync with xterm.js' size.

> I am not aware of concepts like PTY or TTY...

Not sure what to answer here, beside - _then maybe fix that?_ xterm.js is literally made to run at a PTY, so you wont get far with zero knowledge about that (at least the PTY-master interface mechs, the termios stuff is not important as terminal integrator).

Also see https://github.com/xtermjs/xterm.js/discussions/4488, https://github.com/xtermjs/xterm.js/issues/4434#issuecomment-1482766980, https://github.com/xtermjs/xterm.js/discussions/4436#discussioncomment-5302659
